ZenPacks/zenoss/ZenPackLib/lib/spec/GraphPointSpec.py

```diff
@@ -133,9 +134,14 @@ def thresholdLegends(self, value):
             raise ValueError("thresholdLegends must be specified as a dict or string (got {})".format(value))
         for id, data in self._thresholdLegends.items():
             if not isinstance(data, dict):
-                data = {'legend': None, 'color': None}
+                data = {'legend': None, 'color': None, 'threshId': None}
             self._thresholdLegends[id]['legend'] = data.get('legend')
             self._thresholdLegends[id]['color'] = data.get('color')
+            self._thresholdLegends[id]['threshId'] = data.get('threshId', id)
+
+    def get_threshold_ids(self):
+        """find thresholds related to this datapoint"""
+        return [t.name for t in self.template_spec.template_spec.thresholds.values() if self.dpName in t.dsnames]
```

Comment:
no it's not needed those will always exist -- the template_specs are required for this class' existence, and the 'thresholds' will always be an OrderedDict even if empty. -- just checked to be sure